packages/liveblocks-emails/src/comment-body.tsx

```diff
@@ -0,0 +1,345 @@
+                key={`lb-comment-body-mention-${inlineIndex}`}
+                element={inline}
+                user={resolvedUsers.get(inline.id)}
+              />
+            ) : null;
+          }
+
+          if (isCommentBodyLink(inline)) {
+            const href = toAbsoluteUrl(inline.url) ?? inline.url;
+            return (
+              <Components.Link
+                key={`lb-comment-body-link-${inlineIndex}`}
+                element={inline}
+                href={href}
+              />
+            );
+          }
+
+          if (isCommentBodyText(inline)) {
+            return (
+              <Components.Text
+                key={`lb-comment-body-text-${inlineIndex}`}
+                element={inline}
+              />
+            );
+          }
+
+          return null;
+        });
+
+        return (
+          <Components.Paragraph key={`lb-comment-body-paragraph-${index}`}>
+            {children}
+          </Components.Paragraph>
+        );
+      }
+      default:
+        console.warn(
+          `Unsupported comment body block type: "${JSON.stringify(block.type)}"`
+        );
+        return null;
+    }
+  });
+
+  return (
+    <Components.Slot key={"lb-comment-body-slot"}>{blocks}</Components.Slot>
+  );
+}
+
+export type ConvertCommentBodyAsHTMLStyles = {
+  /**
+   * The default inline CSS styles used to display paragraphs.
+   */
+  paragraph: InlineCSSString;
+  /**
+   * The default inline CSS styles used to display text `<strong />` elements.
+   */
+  strong: InlineCSSString;
+  /**
+   * The default inline CSS styles used to display text `<code />` elements.
+   */
+  code: InlineCSSString;
+  /**
+   * The default inline CSS styles used to display links.
+   */
+  mention: InlineCSSString;
+  /**
+   * The default inline CSS styles used to display mentions.
+   */
+  link: InlineCSSString;
+};
+
+const baseStyles: ConvertCommentBodyAsHTMLStyles = {
+  paragraph: "font-size:14px;",
+  strong: "font-weight:500;",
+  code: 'font-family:ui-monospace, Menlo, Monaco, "Cascadia Mono", "Segoe UI Mono", "Roboto Mono", "Oxygen Mono", "Ubuntu Mono", "Source Code Pro", "Fira Mono", "Droid Sans Mono", "Consolas", "Courier New", monospace;background-color:rgba(0,0,0,0.05);border:1px solid rgba(0,0,0,0.1);border-radius:4px;',
+  mention: "color:blue;",
+  link: "text-decoration:underline;",
+};
+
+/** @internal */
+const getCommentBodyAsHTMLStyles = (
+  styles: Partial<ConvertCommentBodyAsHTMLStyles> = {}
+): ConvertCommentBodyAsHTMLStyles => {
+  return {
+    paragraph: styles.paragraph
+      ? sanitizeInlineCSS(styles.paragraph)
+      : baseStyles.paragraph,
+    strong: styles.strong
+      ? sanitizeInlineCSS(styles.strong)
+      : baseStyles.strong,
+    code: styles.code ? sanitizeInlineCSS(styles.code) : baseStyles.code,
+    mention: styles.mention
+      ? sanitizeInlineCSS(styles.mention)
+      : baseStyles.mention,
+    link: styles.link ? sanitizeInlineCSS(styles.link) : baseStyles.link,
```

Comment:
Nice catch @marcbouchenoire 🙏🏻 
You're absolutely right here. I shouldn't replace but I should merge.

I will try to think out a solution to do so gracefully.

Thanks 🙏🏻

Reply:
I like it, plus it seems much friendlier to use than dealing with raw strings 👍

Reply:
It might look confusing to use something from React in the non-React function of the two imo so I'd be in favor of duplicating the `@types/react` [source](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0cd318f955664bafd6fd67b37a697f76dd62fac7/types/react/index.d.ts#L2585-L2594) which does use `csstype` but it's only during development and `@types/react` already uses it so it won't add anything

How do you plan to do the object → string conversion? Can we still use something native like `CSSStyleDeclaration`?

Reply:
Yeah that's why I thought.

I thought about using a derived type like this 👇🏻 
```ts
export type CSSStyleProperties = Omit<
  CSSStyleDeclaration,
  | "getPropertyPriority"
  | "getPropertyValue"
  | "item"
  | "removeProperty"
  | "setProperty"
  | "length"
  | "parentRule"
  | number
>;
```
to act as a configuration object type where all values will be strings.  Then during the conversion I will just have to camelcase the keys without touching the values or converting them like for example `{fontSize: 14}` → `font-size:14px;`.

It's the best trade I have for now in mind without adding complexity on everything.
WDYT?

Reply:
I was wondering if we could use something native because there are edge cases like `{ WebkitFilter: "blur(20px)" }` becoming `-webkit-filter: blur(20px)` to think about. I'm going to check React's source for this since I don't think they keep track of CSS properties/prefixes.

Reply:
-- For Continuity & Legacy --
After our huddle sync w/ Marc yesterday we decided to well doing for an object definition based the type `Properties` from `csstype`. 

So then we will just have to convert the object as an inline css string. `csstype` is added as dev dependency of `@liveblocks/emails` on a fixed version (same as the one used in React).

For now it's the best choice to make as it will give flexibility and auto-completion to developers. And as it's dev dependency it will not make the developer's bundle more heavy.